actions/lib/prepare_archive_package.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Simple script that should be executed on a biotank when we want to prepare a
+# runfolder to be archived to PDC via TSM.
+
+set -e
+
+# The full path to the runfolder that we want to archive.
+RUNFOLDER=$1
+
+RUNFOLDER_NAME=$(basename ${RUNFOLDER})
+
+# The exclude pattern is configurable upstreams, but it should hopefully look
+# something like the following. Escaping various quotes can be messy though. 
+# EXCLUDE="^./Config|^./InterOp|^./SampleSheet.csv|^./Unaligned|^./Data|^./Thumbnail_Images"
+EXCLUDE=$2
+
+cd ${RUNFOLDER}
+
+# TODO: Shall we upload it when it exists?
+if [ -f ./${RUNFOLDER_NAME}.tar.gz ]; then
+  echo "Gziped archive file ${RUNFOLDER_NAME}.tar.gz already exists. Manual intervention required."
+  exit 1
+fi
+
+# Create list of files to include in the compressed part of the archive.
+# Should have a couple of file patterns excluded from the config.
+# Note that we're only adding the files, and not the directories, because
+# it would cause tar to return with a failure code when removing directories
+# before files in subdirs have been removed. (And if we use the exclude features
+# from tar instead then it will always add "." which causes its own problems
+# as it can't be removed.)
+find . -type f -print | egrep -v ${EXCLUDE} > ./.files-to-pack
+
+# Pack everything together and then remove them at the same time. More
+# efficiently space wise than removing them afterwards. Compressing is perhaps
+# not strictly necessary, but it will save some space for primarily the log files.
+tar -T ./.files-to-pack -c -z -f ./${RUNFOLDER_NAME}.tar.gz --remove-files
```

Comment:
I'm wondering if removing files as they are added to the tarball in combination with the error message above if the tarfile already exists, will increase the risk of accidentally removing data (e.g. removing a half-baked tarfile as a cleanup step before re-running the packaging step)?